plugin/src/main/resources/io/jenkins/plugins/remotingkafka/kubernetes/kafka-statefulset.yaml

```diff
@@ -0,0 +1,66 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: kafka
+spec:
+  selector:
+    matchLabels:
+      app: kafka
+  serviceName: kafka-svc-hs
+  replicas: 1
+  updateStrategy:
+    type: RollingUpdate
+  podManagementPolicy: Parallel
+  template:
+    metadata:
+      labels:
+        app: kafka
+    spec:
+      containers:
+        - name: kafka
+          image: wurstmeister/kafka:2.12-2.2.0
+          env:
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: POD_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+            - name: KAFKA_ZOOKEEPER_CONNECT
+              value: "zookeeper-svc:2181"
+            - name: KAFKA_LISTENERS
+              value: "INTERNAL://:9092,EXTERNAL://:9094"
+            - name: KAFKA_INTER_BROKER_LISTENER_NAME
+              value: "INTERNAL"
+            - name: KAFKA_LISTENER_SECURITY_PROTOCOL_MAP
+              value: "INTERNAL:PLAINTEXT,EXTERNAL:PLAINTEXT"
+            - name: KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR
+              value: "1"
+          command:
+            - sh
+            - -exc
+            - |
+              export KAFKA_BROKER_ID=${HOSTNAME##*-} 
+              export KAFKA_ADVERTISED_LISTENERS="${KAFKA_ADVERTISED_LISTENERS},INTERNAL://${POD_NAME}.kafka-svc-hs.${POD_NAMESPACE}:9092"
+              echo "Waiting for Zookeeper"
+              while true; do
+                if [ "imok" = "$(echo ruok | nc -w 1 zookeeper-svc 2181)" ]; then
+                  break;
+                fi
+                sleep 2
+              done
+              exec start-kafka.sh
+          volumeMounts:
+            - name: kafka-data
+              mountPath: /kafka
+  volumeClaimTemplates:
+    - metadata:
+        name: kafka-data
+      spec:
+        accessModes: [ "ReadWriteOnce" ]
+        storageClassName: standard
+        resources:
+          requests:
+            storage: 2Gi
```

Comment:
Should you added a readiness probe to kafka container?

Reply:
@afalko The examples I saw often use `tcpSocket` to determine Kafka's readiness. However in this case, I couldn't think of a reliable way to check for Kafka container's readiness, because it depends on Zookeeper's, and Zookeeper depends on the internet connection to pull images which is very uncertain.

Reply:
You can make readiness probe a script executed on container. I think to start, you can check if port 9092 comes up and later, we'll want to check a metric called "under replicated partitions" (URP). We can add a TODO for now and handle this in another PR.